airbyte-cdk/python/airbyte_cdk/sources/file_based/schema_helpers.py

```diff
@@ -170,8 +183,51 @@ def conforms_to_schema(record: Mapping[str, Any], schema: Mapping[str, str]) ->
     return True
 
 
-def type_mapping_to_jsonschema(type_mapping: Mapping[str, Any]) -> Mapping[str, str]:
+def _parse_json_input(input_schema: Optional[Union[str, Dict[str, str]]]) -> Optional[Mapping[str, str]]:
+    try:
+        schema = json.loads(input_schema)
+        if not all(isinstance(s, str) for s in schema.values()):
+            raise ConfigValidationError(
+                FileBasedSourceError.ERROR_PARSING_USER_PROVIDED_SCHEMA, details="Invalid input schema; nested schemas are not supported."
+            )
+
+    except json.decoder.JSONDecodeError:
+        return None
+
+    return schema
+
+
+def type_mapping_to_jsonschema(input_schema: Optional[Union[str, Mapping[str, str]]]) -> Optional[Mapping[str, str]]:
     """
     Return the user input schema (type mapping), transformed to JSON Schema format.
+
+    Verify that the input schema:
+        - is a key:value map
+        - all values in the map correspond to a JsonSchema datatype
     """
-    ...
+    if not input_schema:
+        return None
+
+    result_schema = {}
+
+    json_mapping = _parse_json_input(input_schema)
```

Comment:
json_mapping can return `None`. We should either null check its output or have it return an empty dict instead